File: Ruby/basics/janken.rb
# じゃんけんプログラム

# 標準入力から1行取得
line = gets.chomp
# カンマで分割して、配列に代入
janken = line.split(",")
# 配列の要素数を変数に代入
num = janken.length
# 配列の中身を出力
p janken
# ランダムに選んだ配列の要素を出力
p janken[rand(num)]

File: java/basic/primer4/Janken.java
package java.basic.primer4;

import java.util.Scanner;

public class Janken {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    // 入力取り込み
    String line = sc.nextLine();
    // ","区切りで配列に格納
    String[] janken = line.split(",");
    // 乱数の生成
    double rand = Math.random() * janken.length;
    // 表示
    System.out.println(janken[(int) rand]);
    sc.close();
  }
}
